keps/sig-auth/3257-trust-anchor-sets/README.md

```diff
@@ -0,0 +1,1184 @@
+Longer term, we may want to require automated upgrade/rollback tests, but we
+are missing a bunch of machinery and tooling and can't do that now.
+-->
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+<!--
+Even if applying deprecation policies, they may still surprise some users.
+-->
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+Use of existing metrics to track operations against certificates.k8s.io/v1alpha1 ClusterTrustBundle objects will provide a good proxy for the number.
+
+Kubelet should export a count of pemTrustAnchors volumes mounted, and a latency histogram for time spent specifically in the pemTrustAnchors volume mount process.
+
+###### How can someone using this feature know that it is working for their instance?
+
+<!--
+For instance, if this is a pod-related feature, it should be possible to determine if the feature is functioning properly
+for each individual pod.
+Pick one more of these and delete the rest.
+Please describe all items visible to end users below with sufficient detail so that they can verify correct enablement
+and operation of this feature.
+Recall that end users cannot usually observe component logs or access metrics.
+-->
+
+- [ ] Events
+  - Event Reason:
+- [ ] API .status
+  - Condition name:
+  - Other field:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+Operations on ClusterTrustBundle objects are covered by the existing [API call latency SLOs](https://github.com/kubernetes/community/blob/master/sig-scalability/slos/api_call_latency.md).
+
+Usage of pemTrustAnchors projected volumes are covered by the existing [pod startup latency SLOs](https://github.com/kubernetes/community/blob/master/sig-scalability/slos/pod_startup_latency.md).
+
+We may need a new SLO to govern the latency between an update to a ClusterTrustBundle's certificate set and that change being made available to pods using pemTrustAnchors volumes.
+
+We may need a new SLO to govern the latency between an update to a ClusterTrustBundle's certificate set and that change being made respected by webhook and aggregated API server connections.
+
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+<!--
+Pick one more of these and delete the rest.
+-->
+
+- [ ] Metrics
+  - Metric name:
+  - [Optional] Aggregation method:
+  - Components exposing the metric:
+- [ ] Other (treat as last resort)
+  - Details:
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+If we can find a low-cardinality way for Kubelet to report what object generation of each ClusterTrustBundle object that it is currently exporting to pods, that would help implement the two new SLOs proposed above.
```

Comment:
I think you can have a count of volume updates and a count of mount errors even if you don't subdivide by clustertrustbundle.  That would inform you of feature problems on a per-kubelet basis.

Reply:
I've documented that in the metrics section.